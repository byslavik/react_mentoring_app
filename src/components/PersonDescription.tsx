import * as React from 'react';

import { connect } from 'react-redux';

import {getFilmsByQuery} from '../actions/films';

import {urlParams} from './../models/common.models'; 

import * as s from './scss/FilmDescripiton.scss';

import {FullFilm, Film} from '../models/film.model';
import {Person, FullPerson} from '../models/person.model';
import {ReduxStore} from './../models/redux.model'; 
import {SETTINGS} from './../settings';

interface PersonProps {
  currentFilm?: FullPerson;
  getFilmsByQuery?: any;
  match?:any
}



export class PersonDescription extends React.PureComponent <PersonProps, any> {
    componentWillMount() {
      this.getItems(this.props.match.params.id)
    }
    componentWillReceiveProps(nextProps:any){
      if(this.props.match.params.id != nextProps.match.params.id) {
        this.getItems(nextProps.match.params.id)
      }
    }
    getItems(id:string) {
      
      let params:urlParams = {
        type: "person",
        query: id
      }
      this.props.getFilmsByQuery(params);
    }
    render() {

      const personData = this.props.currentFilm;
      
        return (<article className={s.filmContainer} key={this.props.match.params.id}>
          <div className={s.imgWrap}>
            <img src={SETTINGS.imgUrl + personData.profile_path} alt={personData.name}/>
          </div>
          <div className={s.filmDescription}>
            <h1>{personData.name} <span>{(Math.round(personData.popularity * 10)/10)}</span></h1>
            <p className={s.filmSummary}>{personData.biography}</p>
          </div>
        </article>)
    }
}

export default connect<any, any, any>(
  (state: ReduxStore) => ({
    currentFilm: state.films.currentFilm
  }),
  (dispatch: any) => ({
    getFilmsByQuery: (query:urlParams)=> {
      dispatch({type: 'GET_FILM', payload: {dispatchType: 'GET_FILM', urlParams: query}});
    }
   })
)(PersonDescription);