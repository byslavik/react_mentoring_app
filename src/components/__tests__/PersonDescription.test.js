import * as React from 'react';
import {PersonDescription} from '../PersonDescription';
import * as renderer from 'react-test-renderer';
import {filmData, match} from './testData'


describe('PersonDescription component', ()=>{
  it('PersonDescription component shows correctly', () => {
    const tree = renderer.create(
      <PersonDescription currentFilm={filmData} match={match} getFilmsByQuery={jest.fn()}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})