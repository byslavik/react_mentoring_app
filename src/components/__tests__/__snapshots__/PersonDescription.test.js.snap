// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDescription component PersonDescription component shows correctly 1`] = `
<article
  className={undefined}
>
  <div
    className={undefined}
  >
    <img
      alt="Name"
      src="https://image.tmdb.org/t/p/w500//somepath.jpg"
    />
  </div>
  <div
    className={undefined}
  >
    <h1>
      Name
       
      <span>
        100500
      </span>
    </h1>
    <p
      className={undefined}
    />
  </div>
</article>
`;
